MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -246,62 +297,251 @@ int main(int argc, char * argv[])
+                //{
+                //    std::wcout << std::endl;
+                //    std::wcout << "Applying ACLs is not applicable for CIM files." << std::endl;
+                //    std::wcout << std::endl;
+                //    return E_INVALIDARG;
+                //}
+                if (!EndsWith(unpackDestination, L".cim"))
+                {
+                    std::wcout << std::endl;
+                    std::wcout << "Invalid CIM file name." << std::endl;
+                    std::wcout << std::endl;
+                    return E_INVALIDARG;
+                }
+
+                // Create a temporary directory to unpack package(s) since we cannot unpack to the CIM directly.
+                std::wstring currentDirectory = std::filesystem::current_path();
+                GUID uniqueId;
+                RETURN_IF_FAILED(createGUID(&uniqueId));
+                RPC_STATUS status = UuidCreate(&uniqueId);
+                if (status != RPC_S_OK && status != RPC_S_UUID_LOCAL_ONLY)
+                {
+                    return HRESULT_FROM_WIN32(status);
+                }
+                RPC_WSTR uniqueIDRPCString = NULL;
+                std::wstring uniqueIDString;
+                if (UuidToStringW(&uniqueId, &uniqueIDRPCString) == RPC_S_OK)
+                {
+                    uniqueIDString = (WCHAR*) uniqueIDRPCString;
+                    RpcStringFreeW(&uniqueIDRPCString);
+                }
+                std::wstring tempDirPathString = currentDirectory + L"\\" + uniqueIDString;
+                std::filesystem::path tempDirPath(tempDirPathString);
+
+                try
+                {
+                    bool createTempDirResult = std::filesystem::create_directory(tempDirPath);
+                    // Since we're using a GUID, this should almost never happen
+                    if (!createTempDirResult)
+                    {
+                        std::wcout << std::endl;
+                        std::wcout << "Failed to create temp directory " << tempDirPathString << std::endl;
+                        std::wcout << "This may occur when the directory path already exists. Please try again."  << std::endl;
+                        std::wcout << std::endl;
+                        return E_UNEXPECTED;
+                    }
+                }
+                catch (std::exception& e)
+                { 
+                    // Again, we expect that the creation of the temp directory will fail very rarely. Output the exception
+                    // and have the user try again.
+                    std::wcout << std::endl;
+                    std::wcout << "Creation of temp directory " << tempDirPathString << " failed with error: " << e.what() << std::endl;
+                    std::wcout << "Please try again." << std::endl;
+                    std::wcout << std::endl;
+                    return E_UNEXPECTED;
+                }
+
+                // should applying acls be true by default?

```

Comment:
>// should applying acls be true by default? [](start = 16, length = 43)

shouldn't it be what the user passes in? i would think in most cases the user would want to apply ACLs though, right? not sure what the usage scenarios around this are.